.github/workflows/test.yml

```diff
@@ -1,217 +1,178 @@
-name: Cargo tests
+name: Cargo Tests
+
 on:
   push:
-    branches:
-      - main
+    branches: [ main ]
   pull_request:
+
+
 jobs:
   clippy:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions-rs/toolchain@v1
-        with:
-            components: clippy
-            override: true
-      - name: Install dependencies
-        run: sudo apt install -y openssl libkrb5-dev
-      - uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --features=all
-
-  format:
-    runs-on: ubuntu-latest
+    - uses: actions/checkout@v5 # checkout versions have been updated. previous v1/2
+    - uses: dtolnay/rust-toolchain@stable # changed to stable
+      with:
+        components: rustfmt, clippy
 
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          components: rustfmt
-          override: true
-      - uses: mbrobbel/rustfmt-check@master
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y libkrb5-dev krb5-user && sudo ln -s /usr/include/krb5/gssapi /usr/include/gssapi
+
+    - run: cargo clippy --features=all
+
+    - run: cargo fmt --check
 
   cargo-test-linux:
     runs-on: ubuntu-latest
-
     strategy:
       fail-fast: false
       matrix:
-        database:
-          - 2017
-          - 2019
-          - 2022
-          - azure-sql-edge
-        features:
-          - "--features=all"
-          - "--no-default-features"
-          - "--no-default-features --features=chrono"
-          - "--no-default-features --features=time"
-          - "--no-default-features --features=rustls"
-          - "--no-default-features --features=vendored-openssl"
-
-    env:
-      TIBERIUS_TEST_CONNECTION_STRING: "server=tcp:localhost,1433;user=SA;password=<YourStrong@Passw0rd>;TrustServerCertificate=true"
-      RUSTFLAGS: "-Dwarnings"
+        engine: [ 2017, 2019, 2022, "azure" ]
 
     steps:
-      - uses: actions/checkout@v2
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y libkrb5-dev krb5-user && sudo ln -s /usr/include/krb5/gssapi /usr/include/gssapi
+
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@stable
 
-      - uses: actions-rs/toolchain@v1
+    - name: Setup Cargo build cache
+      uses: actions/cache@v4
+      with:
+        path: target
+        key: ubuntu-cargo
```

Comment:
this is correct, but perhaps we should leave a comment that we use the same cache key irrespective of the SQL server version